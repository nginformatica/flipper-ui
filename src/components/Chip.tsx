import { Chip as MuiChip, withStyles } from '@material-ui/core'
import React from 'react'

interface IProps {
    style?: object
    label: string
    clickable?: boolean
    color?: 'default' | 'primary' | 'secondary'
    avatar?: React.ReactElement<any>
    deleteIcon?: React.ReactElement<any>
    onDelete?: (value) => void
}

const styles = theme => ({
    root: {
        borderRadius: '20px',
        height: '40px'
    }
})

const Chip = ({ color = 'primary', ...otherProps }: IProps) =>
    <MuiChip color={ color } { ...otherProps } />

export default withStyles(styles)(Chip)
