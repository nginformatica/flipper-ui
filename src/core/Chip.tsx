import { Chip as MuiChip } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import React, { FC } from 'react'
import { DefaultProps } from './types'
import { ChipProps } from '@material-ui/core/Chip'

export interface IChipProps extends DefaultProps {
    square?: boolean
}

const useStyles = makeStyles({
    root: {
        borderRadius: '4px'
    },
    avatar: {
        borderRadius: '2px'
    }
})

const Chip: FC<ChipProps & IChipProps> = ({
    square,
    padding,
    margin,
    style = {},
    ...otherProps
}) => {
    const classes = useStyles()

    return (
        <MuiChip
            classes={square ? classes : undefined}
            style={{ padding, margin, ...style }}
            {...otherProps}
        />
    )
}

export default Chip
