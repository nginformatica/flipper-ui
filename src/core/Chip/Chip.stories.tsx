import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import Chip from '.'
import Avatar from '../Avatar'
import FaceIcon from '@material-ui/icons/Face'

export default {
    title: 'Chip',
    component: Chip,
    argTypes: {
        backgroundColor: { control: 'color' }
    }
} as ComponentMeta<typeof Chip>

const Template: ComponentStory<typeof Chip> = args => <Chip {...args} />

const commonArgs = {
    label: 'Hello darkness my old friend',
    children: 'Chip'
}

export const Default = Template.bind({})
Default.args = {
    ...commonArgs
}

export const Squared = Template.bind({})
Squared.args = {
    ...commonArgs,
    square: true
}

export const WithInitialsAvatar = Template.bind({})
WithInitialsAvatar.args = {
    ...commonArgs,
    avatar: <Avatar>HD</Avatar>
}

export const WithIconAvatar = Template.bind({})
WithIconAvatar.args = {
    ...commonArgs,
    avatar: (
        <Avatar>
            <FaceIcon />
        </Avatar>
    )
}

export const WithImageAvatar = Template.bind({})
WithImageAvatar.args = {
    ...commonArgs,
    avatar: <Avatar src='https://picsum.photos/40' />
}

export const SecondaryColor = Template.bind({})
SecondaryColor.args = {
    ...commonArgs,
    color: 'secondary'
}
