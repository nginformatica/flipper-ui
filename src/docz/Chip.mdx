---
name: Chip
route: /chip
---

import Avatar from '../components/Avatar'
import Chip from '../components/Chip'
import FaceIcon from '@material-ui/icons/Face'
import { Playground, PropsTable } from 'docz'

# Chip

## Properties
<PropsTable of={ Chip } />

## Without Avatar
<Playground>
    <Chip label='Hello darkness my old friend' />
</Playground>

## With Initials Avatar
<Playground>
    <Chip
        label='Hello darkness my old friend'
        avatar={ <Avatar>HD</Avatar> }
    />
</Playground>

## With Icon Avatar
<Playground>
    <Chip
        label='Hello darkness my old friend'
        avatar={ <Avatar><FaceIcon /></Avatar> }
    />
</Playground>

## With Image Avatar
<Playground>
    <Chip
        label='Hello darkness my old friend'
        avatar={
            <Avatar
                src="https://picsum.photos/40"
            />
        }
    />
</Playground>

## Secondary color
<Playground>
    <Chip label='Hello darkness my old friend' color='secondary' />
</Playground>
