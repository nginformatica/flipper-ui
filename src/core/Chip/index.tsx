import { Chip as MuiChip } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import React from 'react'
import { DefaultProps } from '../types'
import { ChipProps } from '@material-ui/core/Chip'

export interface IChipProps extends DefaultProps {
    square?: boolean
}

const useStyles = makeStyles({
    root: {
        borderRadius: '4px'
    },
    avatar: {
        borderRadius: '2px'
    }
})

const Chip = ({
    square,
    padding,
    margin,
    style = {},
    ...otherProps
}: ChipProps & IChipProps) => {
    const classes = useStyles()

    return (
        <MuiChip
            classes={square ? classes : undefined}
            style={{ padding, margin, ...style }}
            {...otherProps}
        />
    )
}

export default Chip
