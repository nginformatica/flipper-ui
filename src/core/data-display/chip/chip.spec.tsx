import * as React from 'react'
import { render, screen, waitFor } from '@testing-library/react'
import Chip from '.'

describe('Chip', () => {
    it('should render', () => {
        render(<Chip label='chip-label' onChange={jest.fn()} />)

        const label = screen.getByText('chip-label')

        expect(label).toBeDefined()
    })

    it('should render square', () => {
        render(<Chip square label='chip-label' onChange={jest.fn()} />)

        const label = screen.getByText('chip-label')

        const findPrimaryClass = Array.from(
            (label.parentElement || label).classList
        )
            .join(' ')
            .indexOf('makeStyles-root-')

        expect(findPrimaryClass).toBeGreaterThan(-1)
    })

    it('should call onDelete', () => {
        const onDelete = jest.fn()

        render(<Chip label='chip-label' onDelete={onDelete} />)

        const svg = screen.getByRole('button')

        svg.click()

        waitFor(() => {
            expect(onDelete).toHaveBeenCalled()
        })
    })
})
