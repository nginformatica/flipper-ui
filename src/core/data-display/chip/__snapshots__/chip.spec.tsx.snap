// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip should match snapshot 1`] = `
<div>
  <div
    class="MuiChip-root"
  >
    <span
      class="MuiChip-label"
    >
      chip-label
    </span>
  </div>
</div>
`;
